File: Reqnroll/Chapter13/13.1-TheLocalScenarioContextPattern/Before/WIMP.LocalScenarioContextSample.App/Services/AuthenticationService.cs
using WIMP.LocalScenarioContextSample.App.Infrastructure;

namespace WIMP.LocalScenarioContextSample.App.Services;

public static class AuthenticationService
{
    public static void Login(string customerName)
    {
        UserSession.Current.LoggedInCustomerName = customerName;
    }

    public static bool IsLoggedIn(string? customerName)
    {
        return customerName != null && UserSession.Current.LoggedInCustomerName == customerName;
    }
}

File: CucumberJVM/Chapter15/15.7-StubDependencyPattern/Before/src/main/java/com/wimp/app/services/AuthenticationService.java
package com.wimp.app.services;

import com.wimp.app.data.InMemoryDataStore;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;
import org.springframework.web.server.ResponseStatusException;

@Service
public class AuthenticationService {
    private final InMemoryDataStore dataStore;

    public AuthenticationService(InMemoryDataStore dataStore) {
        this.dataStore = dataStore;
    }

    public String login(String customerName, String password) {
        if (!dataStore.validatePassword(customerName, password)) {
            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "Invalid credentials");
        }
        return "token-" + customerName;
    }
}
